modules/httpboot/httpboot_api.rb

```diff
@@ -0,0 +1,10 @@
+class Proxy::HttpbootApi < Sinatra::Base
+  helpers ::Proxy::Helpers
+
+  get "/*" do
+    file = Pathname.new(params[:splat].first).cleanpath
+    real_file = Pathname.new(File.join(Proxy::TFTP::Plugin.settings.tftproot, file)).realpath
```

Comment:
Should this be `Proxy::Httpboot::Plugin.settings.root_dir` instead?

Reply:
Also, if you get /EFI or a non-existent file, we return 500.  It'd be nice if we return the right http error codes, 404, 503, etc.

Reply:
Good catch, yeah absolutely 404. Changing.

Reply:
So I am implementing extra verbose checks just to be sure.